prov/util/src/util_ns.c

```diff
@@ -0,0 +1,505 @@
+/*
+ * Copyright (c) 2017 Intel Corporation. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*
+ * A name server is started on each
+ * node as a thread within one of the processes on that node. It maintains
+ * a database that maps "services" to "endpoint names". Other processes
+ * on the same node talk to this name server to update mapping information.
+ *
+ * To resolve a "node:service" pair into an provider internal endpoint name
+ * that can be used as the input of fi_av_insert, a process needs to make
+ * a query to the name server residing on "node".
+ */
+
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netdb.h>
+
+#include <fi_util.h>
+#include <rdma/providers/fi_log.h>
+#include <fi.h>
+
+#include "rbtree.h"
+
+#define OFI_UTIL_NS_DEFAULT_PORT	5678
```

Comment:
We should drop the default port number from the generic name service code, and have this set by the providers.  The providers can use env vars to override their defaults.  Then pass in the port as part of ns_start().